### PR TITLE
symbolic intents

### PR DESCRIPTION
**Pull request by davidcarlisle**
as discussed in #448 this PR adds a new `symbol` class for intent along with a warning that perhaps it shouldn't be used until software catches up.

### CODE REVIEW

**Comment by davidcarlisle** [comment]
rendered view at https://davidcarlisle.github.io/mathml/#mixing_intent_grammar

**Comment by NSoiffer** [comment]
Two things:
1. I think it would be better to omit the space before the property as then it is clearer that the property belongs to what precedes as opposed to sitting on its own. I understand space is allowed -- this is for clarity sake for those first seeing the spec.
2. The discussion for why literal strings are bad does not get at what to me is the problem with them: they don't take in the needs of the reader. I think this can be made clearer by adding to `semi-factorial:postfix($x)` saying it might be read as "x semi factorial" by if "x" were a simple variable, but a blind user needs to know where something starts and ends. A more complex expression such as `x+2`, then it might be read as "quantity x plus 2 end semi-factorial". And then reference this when talking about problems with literals (e.g., if `_semi-factorial` had been used). This may not be a good example though because someone would probably use parens for "x+2", so some bracketed quantity such as "norm" is better.  But that's not postfix.  (Also, `$n` is maybe better for the `semi-factorial` example)

**Comment by davidcarlisle** [comment]
@NSoiffer 

> I think it would be better to omit the space before the property as then it is clearer that the property belongs to what precedes as opposed to sitting on its own. I understand space is allowed -- this is for clarity sake for those first seeing the spec.

Do you mean omit the `S` in the grammar,  or mean in examples do `foo:infix` not  `foo :infix` ?

> The discussion for why literal strings are bad does not get at what to me is the problem with them:

Yes I agree but that text isn't touched by this PR (or the one just merged that combined `@`and `:`) so it could be separately editorially improved on `main` or a separate PR.

**Comment by NSoiffer** [comment]
The later: `foo:infix` in the examples. At first glance, the `:infix` is not attached to anything. `foo:infix` makes it much clearer that "infix" is attached to "foo".

**Comment by davidcarlisle** [comment]
ah OK then, I have no objection to changing the examples in the spec, but there are no  examples using `:`  in the text added by this PR,

**Comment by brucemiller** [comment]
I'm not sure I see the advantage of adding this `symbol` clause. Only adding math symbols makes sense on the surface, but only slightly reduces the Error footprint. The caveat implies that the AT may or may not actually look at the symbol but may (or may not?) pass it directly on to some underlying speech-engine layer.  I wonder if anyone who hasn't participated in these discussions will understand that distinction.

I suspect I'd prefer sticking to more restrictive literals (if I had to choose).

File: src/mixing.html [hunk 1]
```diff
@@ -81,7 +82,9 @@ <h4 id="mixing_intent_grammar">The Grammar for <code class="attribue">intent</co
    </pre>
 
    <p>Here <a href="https://www.w3.org/TR/REC-xml-names/#NT-NCName"><code>NCName</code></a>
-   is as defined in  in [[xml-names]], and <code>digit</code> is a character in the range 0–9.</p>
+   is as defined in  in [[xml-names]],
+   <code>\pNd</code> is a character of class Nd (decimal digit),
```
Comment (dginev) [inline src/mixing.html:86]: was `\pNd` meant to replace `\d` in the `number` rule? It hasn't yet.

**Comment by davidcarlisle** [comment]
@brucemiller 
> I suspect I'd prefer sticking to more restrictive literals (if I had to choose).

yes I might too, not sure yet, but I made the PR to have something concrete to shoot down.

I suspect my main use of this  is to enable (or almost enable) one possible explanation of what `$op` means as in  #449 

I'd like to say it somehow causes the implied intent of the element  `op`  to be included.

If the element does not have an `intent`  then the implied  intent could/should be its content which  means `<mo>†</mo>` has an implied intent of  `†` which does not mean that we _have_ to allow this in the literal syntax `<mo intent="†">†</mo>` but it might help the explanations if it were possible to expand all `$argref` and have a valid `intent`.

Or we could instead say the implied intent of `<mo>†</mo>` is its default reading, not its content, so `<mo intent="dagger">†</mo>`  but that makes the expansion of `$argref` not just interpretation of properties and choice of speech, system dependent.

File: src/mixing.html (hunk 1 above)
Comment (davidcarlisle) [reply src/mixing.html:86]: @dginev er yes, or the other way round. I oscillated on what form to use, and seem to have used one and described the other...

Comment (davidcarlisle) [reply src/mixing.html:86]: fixed

**Comment by davidcarlisle** [comment]
This PR served its purpose of providing an explicit proposal and allowing discussion of symbolic intents, but I don't feel any consensus that it's  a good idea, so I'm going to close this.

**Pull request closed without merging**